packages/@o3r/design/src/core/design-token/renderers/design-token-style.renderer.ts

```diff
@@ -98,6 +99,38 @@ export const getTokenSorterByRef: DesignTokenListTransform = (variableSet) => {
   };
 };
 
+/**
+ * Reorganize the Tokens based on a ordered list of regexps.
+ * Each regexp is applied only to last part of the Token name (before key rendering).
+ * @param regExps Ordered list of regular expressions defining the order of the Tokens.
+ * @param applyRendererName Determine if the regexps are apply to the rendered Token key. If `false`, it will be applied to the Token key's name (last part of the Token name).
+ */
+export const getTokenSorterFromRegExpList: (regExps: RegExp[], applyRendererName?: boolean) => DesignTokenListTransform = (regExps, applyRendererName = false) => (_variableSet, options) => {
+
+  const applyRegExp = (token: DesignTokenVariableStructure, regExp: RegExp) => (applyRendererName
+    ? token.getKey(options?.tokenVariableNameRenderer)
+    : token.tokenReferenceName.split(TOKEN_KEY_SEPARATOR).at(-1)!
+  ).match(regExp);
```

Comment:
Maybe add a comment so we don't move back to `Regexp.test(string)` as recommended by our new ESLint recommendation